SPHINXsys/src/shared/particle_dynamics/solid_dynamics/contact_dynamics.h

```diff
@@ -71,6 +74,37 @@ namespace SPH
 			virtual void Interaction(size_t index_i, Real dt = 0.0) override;
 		};
 
+		/**
+		* @class ShellContactDensity
+		* @brief Computing the contact density due to shell contact using a 
+		* 		 surfacic integral being solved by Gauss-Legendre quadrature integration.
```

Comment:
As per problem definition, this is tailored for 2D problems, but in 3D the integral should be performed on a surface patch (so a 3x3 integration). Since the file is in core, this should be mentioned in the documentation of the class as a todo/limitation

Reply:
Good points Fabien.
I will consider them in a new PR, as now all checks have passed and we would like to merge it.